modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java

```diff
@@ -1188,6 +1137,59 @@ private void _validateParameterObjectFieldId(
 		}
 	}
 
+	private void _validateType(
+			ObjectDefinition objectDefinition1,
+			ObjectDefinition objectDefinition2, String name,
+			long parameterObjectFieldId, String type)
+		throws PortalException {
+
+		Set<String> defaultObjectRelationshipTypes =
+			ObjectRelationshipUtil.getDefaultObjectRelationshipTypes();
+
+		if (!defaultObjectRelationshipTypes.contains(type)) {
+			throw new ObjectRelationshipTypeException("Invalid type " + type);
+		}
+
+		if (Objects.equals(
+				type, ObjectRelationshipConstants.TYPE_MANY_TO_MANY) ||
+			Objects.equals(type, ObjectRelationshipConstants.TYPE_ONE_TO_ONE)) {
```

Comment:
Could you please check if  we can remove the `Objects.equals(type, ObjectRelationshipConstants.TYPE_ONE_TO_ONE)` condition?

Reply:
As discussion with @marcelabc, @carolmariaabb and @samuelromeiro. Let's keep the code with oneToOne validations. Add only validation that  not allow to add oneToOne relation via API.